examples/quick-start-with-drink/README.md

```diff
@@ -0,0 +1,85 @@
+# Quick start with `drink` library
+
+This is a quick start guide introducing you to smart contract testing with `drink` library.
+We will see how to write tests for a simple smart contract and make use of `drink`'s features.
+
+## Prerequisites
+
+You only need a Rust installed (see [here](https://www.rust-lang.org/tools/install) for help).
+Drink is developed and tested with stable Rust 1.70 (see [toolchain file](../../rust-toolchain.toml)).
+
+## Dependencies
+
+You only need `drink` library brought into your project:
+```toml
+drink = { version = "0.6" }
+```
+
+See [Cargo.toml](Cargo.toml) for a typical cargo setup of a single-contract projects.
+
+## Writing tests
+
+### Preparing contracts
+
+For every contract that you want to interact with from your tests, you need to create a _contract bundle_, which includes:
+ - built contract artifact (`.wasm` file),
+ - contract transcoder (object based on the `.json` file, responsible for translating message arguments and results).
+
+The recommended way is to use `drink::contract_bundle_provider` macro, which will discover all the contract dependencies (including the current crate, if that is the case) and gather all contract bundles into a single registry.
+
+However, if needed, you can do it manually, by running `cargo contract build` for every such contract, and then, bring the artifacts into your tests.
+For this, you might want to use `drink::ContractBundle` API, which includes `ContractBundle::load` and `local_contract_file!` utilities.
+
+### `drink` test macros
+
+`drink` provides a few macros to write tests for smart contracts:
+ - `#[drink::test]` - which marks a function as a test function (similar to `#[test]`).
+ - `#[drink::contract_bundle_provider]` - which gathers all contract artifacts into a single registry.
+
+While neither is required to write `drink` tests, they make it easier to write and maintain them.
+
+### Writing tests
+
+Your typical test module will look like:
+```rust
+#[cfg(test)]
+mod tests {
+    #[drink::contract_bundle_provider]
+    enum BundleProvider {}
+
+    #[drink::test]
+    fn deploy_and_call_a_contract() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
I wonder whether we should maybe provide an alias for this scary looking error type?

Reply:
it'd feel weird for me to expose an alias for this from drink... I've removed full qualified path so that it looks neater